Fix #844: Remove an invalid assert, add nullptr check

asandryh opened the pull request:
Summary:
Use default values for parameters skip_unique_check_tables and read_free_rpl_tables when they are set to NULL.

Test Plan:
mtr

laurynas-biveinis reviewed (commented):
Why not to include the testcase?

Commit:
Simplify the test

Summary:
Don't use CREATE EVENT.

Test Plan:
rocksdb.rocksdb_read_free_rpl_tables

Commit:
Remove an invalid assert, add nullptr check

Summary:
Use default values for parameters skip_unique_check_tables and
read_free_rpl_tables when they are set to NULL.

Test Plan:
rocksdb.rocksdb_read_free_rpl_tables

asandryh commented:
Added a corresponding test

lth commented on storage/rocksdb/ha_rocksdb.cc:
```diff
@@ -6356,31 +6356,27 @@ void ha_rocksdb::free_key_buffers() {
 #endif
 
-  if (!regex_handler.set_patterns(whitelist)) {
+  if (!regex_handler.set_patterns(whitelist ? whitelist : ".*")) {
```
Is this expected behaviour? I would think that setting to null would mean that no tables are going to skip unique checks?

asandryh replied:
This is a bit confusing, since ".*" is defined as the default value for `skip_unique_check_tables` (https://github.com/facebook/mysql-5.6/blob/fb-mysql-5.6.35/storage/rocksdb/ha_rocksdb.cc#L757). Setting a parameter's value to NULL, where NULL is not an accepted value, should fall back to the default value.

lth commented on mysql-test/suite/rocksdb/t/rocksdb_read_free_rpl_tables.test:
```diff
@@ -0,0 +1,10 @@
+CREATE EVENT ev1 ON SCHEDULE EVERY 1 SECOND DO INSERT INTO t1 VALUES (1);
```
I think the `CREATE EVENT` statement is confusing without a comment referencing the github issue (eg. see `mysql-test/suite/rocksdb/t/update.test`)

It'd also be good if we can simplify the test case, since I think we just need the `SET GLOBAL ...  = NULL` part right?

asandryh replied:
No, `CREATE EVENT` is not needed here, I just kept it the same as reported in #844. Will simplify the test.

lth commented:
Superceded by https://github.com/facebook/mysql-5.6/pull/887
